src/logging.cc

```diff
@@ -1148,13 +1221,50 @@ void LogFileObject::Write(bool force_flush,
   if (base_filename_selected_ && base_filename_.empty()) {
     return;
   }
-
-  if (file_length_ >> 20U >= MaxLogSize() || PidHasChanged()) {
+  struct ::tm tm_time;
```

Comment:
The following block does quite some work. Consider moving it into a separate method and add some comments describing the overall purpose and the non-obvious logic.